pkg/controller/velerobackup/velerobackup_controller.go

```diff
@@ -169,74 +185,124 @@ func (r *ReconcileBackup) createVeleroBackupObject(namespace string) error {
+	nsLogger := addNamespaceLoggerFields(r.logger, cp.Namespace)
+	var err error
 
-		hobj.setChecksumAnnotation()
-		obj := hobj.object
+	if found {
+		err = r.Update(context.TODO(), cp)
+	} else {
+		err = r.Create(context.TODO(), cp)
+	}
 
-		err := r.Update(context.TODO(), obj)
-		if err != nil {
-			hobjLogger.WithError(err).Error("Failed update.")
-			return err // If we error at all, then we should just return. The NS will be requeued in the future.
-		}
+	if err != nil {
+		nsLogger.WithError(err).Error("Failed to create or update CheckPoint object in namespace.")
+		return err
 	}
 
+	// Everything succeeded!
 	return nil
 }
 
-func addNamespaceLoggerFields(logger log.FieldLogger, namespace string) log.FieldLogger {
-	return logger.WithField("namespace", namespace)
-}
-
-func getRuntimeObjects(kclient client.Client, typesToList []runtime.Object, namespace string) ([]runtime.Object, error) {
-	nsObjects := []runtime.Object{}
-
-	for _, t := range typesToList {
-		listObj := t.DeepCopyObject()
-		if err := kclient.List(context.TODO(), listObj, client.InNamespace(namespace)); err != nil {
-			return nil, err
+// calculateChecksum computes the runtime object's hash WITHOUT the status as status is not used to determine if the object has changed.
+// We never return an error from this function as that could lead to a
+// situation where other objects change, but because one of the objects
+// checksum errored, then the other objects won't get backed up.
+// Instead for objects that error, we return a constant checksum value.
+// This function should rarely, if ever, return an error.
+func (r *ReconcileBackup) calculateObjectsChecksumWithoutStatus(objects ...runtime.Object) string {
+	checksums := []string{}
+	for _, object := range objects {
+		var meta *metav1.ObjectMeta
+		var spec interface{}
+		var checksum = ""
+		var err error
+
+		switch t := object.(type) {
+		case *hivev1.ClusterDeployment:
+			meta = &t.ObjectMeta
+			spec = &t.Spec
+		case *hivev1.SyncSet:
+			meta = &t.ObjectMeta
+			spec = &t.Spec
+		case *hivev1.DNSZone:
+			meta = &t.ObjectMeta
+			spec = &t.Spec
+		default:
+			r.logger.Infof("Unknown Type: %T", object)
+			checksum = errChecksum
 		}
-		list, err := meta.ExtractList(listObj)
-		if err != nil {
-			return nil, err
+
+		if checksum != errChecksum {
+			checksum, err = controllerutils.GetChecksumOfObjects(meta, spec)
+			if err != nil {
+				r.logger.WithError(err).Info("error calculating object checksum")
+				checksum = errChecksum
+			}
 		}
 
-		nsObjects = append(nsObjects, list...)
+		checksums = append(checksums, checksum)
 	}
 
-	return nsObjects, nil
+	// This ensures that for the same set of objects we generate the same combined checksum.
+	// Otherwise a different order could cause the checksum to be different every time.
+	sort.Strings(checksums)
+
+	combinedChecksum, err := controllerutils.GetChecksumOfObjects(checksums)
```

Comment:
The effect of creating a checksum is the same, but I think you actually want to do the following here. Or did you mean to call `GetChecksumOfObject`?
```
GetChecksumOfObjects(checksums...)
```